src/main.js

```diff
@@ -133,6 +129,14 @@ export async function createApp({ mountPoint, sdk, layout, pages, input = {}, er
     }
 
     afterSdkInitiated(sdk, summary, cache, local);
+
+    if (callback) {
+        pagesDonePromise
```

Comment:
I guess we should consider just returning undefined here, instead of promise, so that new and old (deprecated) apis don't intersect. What do you think @qubyte ?

Reply:
I was 50:50 on this. You're probably right, but it worried me that the promise resolves at a weird time. I'll make the change.

Reply:
I've pushed some updated code for this.